File: store_agent/src/main/java/com.abc/kafka/Receiver.java
package com.abc.kafka;

import com.abc.dao.DecisionDAO;
import com.abc.dao.UserRequestDAO;
import com.abc.dao.VendorDAO;
import com.abc.model.VendorResponse;
import com.abc.model.VendorResponseModel;
import com.abc.model.rest.DecisionResponse;
import com.abc.model.rest.UserRequest;
import com.google.gson.Gson;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.kafka.annotation.KafkaListener;
import org.springframework.stereotype.Component;

@Component
public class Receiver {

    private final Logger log = LoggerFactory.getLogger(this.getClass());

    @Autowired
    private UserRequestDAO userRequestDAO;

    @Autowired
    private DecisionDAO decisionDAO;

    @Autowired
    private VendorDAO vendorDAO;

    @Autowired
    private Gson gson;

    @KafkaListener(topics = "${store.user.request}")
    public void receiveRequest(String message) {
        log.info("Receive user request='{}'", message);
        userRequestDAO.save(gson.fromJson(message, UserRequest.class));
    }

    @KafkaListener(topics = "${store.vendor.response}")
    public void receiveVendorResponse(String message) {
        log.info("Receive vendor response='{}'", message);
        VendorResponse[] responses = gson.fromJson(message, VendorResponse[].class);
        for (VendorResponse response : responses) {
            vendorDAO.save(new VendorResponseModel(response));
        }
    }

    @KafkaListener(topics = "${store.decision}")
    public void receiveDecision(String message) {
        log.info("Receive decision='{}'", message);
        decisionDAO.save(gson.fromJson(message, DecisionResponse.class));
    }

}

File: store_agent/src/main/java/com.abc/kafka/Receiver.kt
package com.abc.kafka

import com.abc.dao.ScoreDAO
import com.abc.model.VendorResponse
import com.abc.model.rest.DecisionResponse
import com.abc.model.rest.UserRequest
import com.google.gson.Gson
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.kafka.annotation.KafkaListener
import org.springframework.stereotype.Component

@Component
class Receiver {

    @Autowired
    lateinit var scoreDAO: ScoreDAO

    @Autowired
    lateinit var gson: Gson

    @KafkaListener(topics = arrayOf("\${store.user.request}"))
    fun receiveRequest(message: String) {
        println("receiveRequest: $message")
        scoreDAO.saveUserRequest(gson.fromJson(message, UserRequest::class.java))
    }

    @KafkaListener(topics = arrayOf("\${store.vendor.response}"))
    fun receiveVendorResponse(message: String) {
        println("receiveVendorResponse: $message")
        scoreDAO.storeVendorResponse(gson.fromJson(message, Array<VendorResponse>::class.java))
    }

    @KafkaListener(topics = arrayOf("\${store.decision}"))
    fun receiveDecision(message: String) {
        println("receiveDecision: $message")
        scoreDAO.storeDecision(gson.fromJson(message, DecisionResponse::class.java))
    }

}